### PR TITLE
Support rust code generation

### PR DESCRIPTION
**Pull request by chfanghr**
Try it out:

```bash
nix build .#address-book-rust-pb
cp -r $(readlink -f ./result) gen
chmod -R +w ./gen
cd gen
nix shell nixpkgs#cargo nixpkgs#rustc nixpkgs#gcc
cargo build
```

### CODE REVIEW

File: src/rust-proto.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    name = "${rustCrateName}"
+    version = "${rustCrateVersion}"
+    edition = "2021"
+
+    [dependencies]
+    prost = "0.12.0"
+    prost-types = "0.12.3"
+    pbjson-types = "0.6"
+    serde = "1.0"
+    tonic = { version = "0.11", features = ["gzip"] }
```
Comment (bladyjoker) [inline src/rust-proto.nix:30]: Tonic/Prost aren't those different modules for the same thing? Do we want to generate with both modules?

**Review by bladyjoker — APPROVED** [review]
Pretty damn nice! Thanks

Can we move `gen` to `tests/rust-proto-gen`? Perhaps even slap a tests that invokes `rustProto` and diffs against `tests/rust-proto-gen`?

File: src/rust-proto.nix (hunk 1 above)
Comment (bladyjoker) [reply src/rust-proto.nix:30]: Oh, seems like one is for types and tonic for gRPC?

**Comment by chfanghr** [comment]
Actually I shouldn't have committed it lol

File: src/rust-proto.nix (hunk 1 above)
Comment (chfanghr) [reply src/rust-proto.nix:30]: Yep. Prost handles data encoding, tonic is used for grpc client and server.

**Pull request merged**